### PR TITLE
Make possible to resize whole sector with given offset

### PR DESCRIPTION
**Pull request by Hume2**

### CODE REVIEW

File: src/object/tilemap.hpp [hunk 1]
```diff
@@ -67,8 +67,9 @@ class TileMap : public GameObject,
   /** resizes the tilemap to a new width and height (tries to not destroy the
    * existing map)
    */
-  void resize(int newwidth, int newheight, int fill_id = 0);
-  void resize(Size newsize);
+  void resize(int newwidth, int newheight, int fill_id = 0,
+              int xoffset = 0, int yoffset = 0);
+  void resize(Size newsize, Size resize_offset);
```
Comment (tobbi) [inline src/object/tilemap.hpp]: Can't the parameters of this be const Size& as well?

File: src/supertux/sector.hpp [hunk 2]
```diff
@@ -195,7 +195,7 @@ class Sector
   /**
    * resize all tilemaps with given size
    */
-  void resize_sector(Size& old_size, Size& new_size);
+  void resize_sector(Size& old_size, Size& new_size, Size& resize_offset);
```
Comment (tobbi) [inline src/supertux/sector.hpp]: Same here: parameters as const references.

File: src/object/tilemap.cpp [hunk 3]
```diff
@@ -209,8 +213,12 @@ ObjectSettings
 TileMap::get_settings() {
   new_size_x = width;
   new_size_y = height;
+  new_offset_x = 0;
```
Comment (tobbi) [inline src/object/tilemap.cpp]: I don't think you want to reset the new_offset_x and new_offset_y values to 0 here. Otherwise they don't stick.

**Comment by maxteufel** [comment]
@Hume2 It needs to be clear what this is doing and how to use it.

**Comment by tobbi** [comment]
I'm postponing this PR from 0.5.0 to 0.5.1, since this needs better UX and other things.

**Comment by tobbi** [comment]
Since there has been no progress on this PR in a long time, I will close it. Please re-open when there's better UX for this lined up.

**Pull request closed without merging**